src/stream/src/executor/dml.rs

```diff
@@ -120,12 +134,66 @@ impl DmlExecutor {
                                 _ => {}
                             }
                         }
+
+                        current_epoch = barrier.epoch.curr.into();
+
+                        if !active_txn_map.is_empty() {
+                            for (txn_id, (_, mut vec)) in active_txn_map
+                                .drain_filter(|_, (epoch, _)| epoch.0 < barrier.epoch.prev)
+                            {
+                                tracing::warn!("txn_id={}. In case of late `TxnMsg::End`, drain those transaction chunks", txn_id);
+                                for chunk in vec.drain(..) {
+                                    yield Message::Chunk(chunk);
+                                }
+                            }
+                        }
```

Comment:
Is it expected that we have a partial txn here when the epoch advances?

Reply:
> more than two epochs

How do we decide this value? 👀